kips/kip-2.md

```diff
@@ -0,0 +1,462 @@
+2. `UserPot` calculates all the ETH owed to the user based on past attendance
+3. `UserPot` updates the `Storage` contract data for the user, moving events which have ended from the user's event list, and setting their withdrawable balance to 0
+4. `UserPot` sends the total ETH back to the user
+
+## Rationale
+
+The above design was arrived at after considering various alternatives as well as other desirable features that users have asked for. Here are the key points:
+
+- **Singleton user pot is necessary** - because our system redistributes ETH between users (i.e. if you don't show up your ETH gets _given_ to other users) it's necessary to have all ETH for a given event stored in one place, as it would be too expensive gas-wise and time-wise to send ETH amongst the participants of an event. This is why we can't have a `UserPot` instance per user.
+
+- **Reuse of deposit** - because we have a single user pot instance, we can easily reuse the payout from an old event as a deposit for a new event. In the sample implementation below, this is actually what happens when a user RSVP's. Thus, users do not need to withdraw their ETH and send it back in in order to attend future events.
+
+- **Easier overview of user balance** - because all ETH is in one place we can easily give the user an on-chain overview of their total deposits and payouts on Kickback.
+
+- **Upgrade-able contracts** - by making the user pot upgradeable (with admin approval) we allow for flexibility in future and for bug fixing capability.
+
+- **Configurable event incentives** - one of the nice things about this architecture is that the `UserPot` is only responsible for summing up deposits and payouts - actual calculation of what the payout should be is done by the `Event` contract. If we wished to test out new strategies (e.g. sweepstakes) or try using a single `Event` instance for multiple events we could easily do so.
+
+- **Potential revenue opportunity** - with the rise of tools and projects such as rDAI and the DAI savings rate there is increasing opportunity of earning interest from staked assets. By changing the structure to hold our customers money it will increase the overall revenue potential when we integrate with such features in future.
+
+## Risks and and trade-offs
+
+Although gas costs for RSVP'ing have gone up, I think this is justifiable given the improvements in overall user experience, plus the additional flexibility we gain over our current model.
+
+The events emitted by the event contract will all stay the same - meaning our backend doesn't need to change enormously for this. However we will no longer show when a user has withdrawn from an event - there's no need to since withdrawals across events are now batched.
+
+Storing all ETH into a single contract may seem risky, but we can negate this with good testing and auditing of the contract, and by managing admin access using multisig accounts with op-sec supervision.
```

Comment:
Can you describe the  detail of the risk when the admin account is compromised? What can be abused and not been abused by admin abusing the upgradable feature? We can mitigate smart contract risk in combination of having good security audit, bounty , and having smart contract cover but not sure if we have good cover of admin private keys being compromised.